src/dir.rs

```diff
@@ -122,18 +121,12 @@ impl ReadDir {
             }
         };
 
-        // If all blocks in the extent have been processed, move to the
-        // next extent on the next iteration.
-        if self.block_index == u64::from(extent.num_blocks) {
-            self.extent = None;
-            return Ok(None);
-        }
-
         // If a block has been fully processed, move to the next block
         // on the next iteration.
         let block_size = self.fs.0.superblock.block_size;
         if self.offset_within_block >= usize_from_u32(block_size) {
-            self.block_index += 1;
+            self.is_first_block = false;
+            self.block_index = None;
             self.offset_within_block = 0;
```

Comment:
I see that this was here before, but I'm not seeing that case that requires setting this here as well as on line 112?

Reply:
I agree, no need to set it twice. Removed the 2nd one